dGame/dComponents/TriggerComponent.cpp

```diff
@@ -296,48 +305,54 @@ void TriggerComponent::HandleUpdateMission(Entity* targetEntity, std::vector<std
 	// then we need a good way to convert this from a string to that enum
 	if (argArray.at(0) != "exploretask") return;
 	MissionComponent* missionComponent = targetEntity->GetComponent<MissionComponent>();
-	if (!missionComponent) return;
+	if (!missionComponent){
+		Game::logger->Log("TriggerComponent::HandleUpdateMission", "Mission component not found!");
+		return;
+	}
 	missionComponent->Progress(eMissionTaskType::EXPLORE, 0, 0, argArray.at(4));
 }
 
 void TriggerComponent::HandlePlayEffect(Entity* targetEntity, std::vector<std::string> argArray) {
-	// [nameID],[effectID],[effectType],[priority(optional)]
+	if (argArray.size() <= 2) return;
 	int32_t effectID = 0;
 	if (!GeneralUtils::TryParse<int32_t>(argArray.at(1), effectID)) return;
 	std::u16string effectType = GeneralUtils::UTF8ToUTF16(argArray.at(2));
 	float priority = 1;
-	GeneralUtils::TryParse<float>(argArray.at(3), priority);
+	if (argArray.size() == 4) GeneralUtils::TryParse<float>(argArray.at(3), priority);
```

Comment:
== 4 or >= 4?  Would there be a case here for more arguments being passed?
Also the few lines above this, are they intending to access argArray.at(0) and 1?  They check for a size of 2 but then access elements 2 and 3

Reply:
`[nameID],[effectID],[effectType],[priority(optional)]` is the arg format
So the last one is optional, and if `size <= 2` it returns, because there are 3 required params. I can do `size < 3` to make that a bit clearer, just me being weird lol

Reply:
index 0 is already in the right type so I just put it straight in the GM call, but gotta convert indexes 1 and 2 to the right types, and index 3 as well, if we had 4 args, so the 4th arg would be at the index of 3